crates/ruff_python_parser/src/python.lalrpop

```diff
@@ -366,6 +368,75 @@ LineMagicExpr: ast::Expr = {
     }
 }
 
+HelpEndLineMagic: ast::Stmt = {
+    <location:@L> <e:Expression<"All">> <suffix:("?")+> <end_location:@R> =>? {
```

Comment:
We're a permissive than the original implementation because we would allow whitespace between the expression and the suffix while the IPython implementation doesn't allow it. For example, `foo ?` would be valid in our case but invalid from IPython.

Reply:
It's not just whitespace. It also includes comments, continuation tokens etc. We could use the source text to retrieve the expression range instead and assert whether it contains any whitespace, comments, or continuation tokens (I'm not suggesting that we should do this, but it's a possibility)

Reply:
i'd add @dhruvmanila's comment as code comment

Reply:
Makes sense. I might play around with your idea.

(I'll also add it as a code comment)